tests/core/test_multi_cluster_app.py

```diff
@@ -281,6 +281,50 @@ def test_admin_access(admin_mc, admin_pc, user_factory, remove_resource):
     wait_for_roles_to_be_updated(admin_mc, updated_mcapp, ["cluster-owner"])
 
 
+def test_add_projects(admin_mc, admin_pc, admin_cc, remove_resource):
+    client = admin_mc.client
+    mcapp_name = random_str()
+    temp_ver = "cattle-global-data:library-wordpress-2.1.10"
+    targets = [{"projectId": admin_pc.project.id}]
+    mcapp1 = client.\
+        create_multi_cluster_app(name=mcapp_name,
+                                 templateVersionId=temp_ver,
+                                 targets=targets,
+                                 roles=["project-member"])
+    wait_for_app(admin_pc, mcapp_name, 60)
+    remove_resource(mcapp1)
+    p = client.create_project(name='test-' + random_str(),
+                              clusterId=admin_cc.cluster.id)
+
+    p = admin_cc.management.client.wait_success(p)
+    remove_resource(p)
+    mcapp1.addProjects(projects=[p.id])
```

Comment:
This action has no safety so if you get a conflict it will not try again. [This](https://github.com/rancher/rancher/blob/master/tests/core/test_pod_security_policies.py#L107) does give you safety for conflict errors but it's not required to use it.